pliers/extractors/text.py

```diff
@@ -313,3 +332,114 @@ def _extract(self, stim):
         features = ['sentiment_' + k for k in scores.keys()]
         return ExtractorResult([list(scores.values())], stim, self,
                                features=features)
+
+
+class SpaCyTokenExtractor(TextExtractor):
+    '''
+    Uses SpaCy to extract features from text. Extracts features for every word(token) in a sentence.
+    Features returned are:
+     text : Verbatim text content.(unicode)
+     lemma_ : Base form of the token, with no inflectional suffixes.(unicode)
+     pos_ : Coarse-grained part-of-speech.(unicode)
+     tag_ : Fine-grained part-of-speech.(unicode)
+     dep_ : Syntactic dependency relation.(unicode)
+     shape_ : Transform of the tokens's string, to show orthographic features. For example, "Xxxx" or "dd".(unicode)
+     is_alpha : Does the token consist of alphabetic characters? (bool)
+     is_stop : Is the token part of a "stop list"? (bool)
+     is_punct : Is the token punctuation?(bool)
+     sentiment : A scalar value indicating the positivity or negativity of the token.(float)
+     is_ascii : Does the token consist of ASCII characters?(bool)
+     is_digit : Does the token consist of digits?(bool)
+    '''
+
+    def __init__(self, model='en_core_web_sm', features=None):
+        verify_dependencies(['spacy'])
+        try:
+            self.model = spacy.load(model)
+
+            print('loaded model: ', self.model)
+        except (ImportError, IOError, OSError) as e:
+            logging.error(
+                "Spacy Models ('{ "+model+" }') not found. Downloading and installing")
+
+            os.system('python -m spacy download '+model)
+            model_na = model[:2]
+            self.model = spacy.load(model_na)
+
+
+        self.features = features
+        super(SpaCyTokenExtractor, self).__init__()
+
+    def _extract(self, stim):
+
+         input_txt = self.model(stim.text)
+         features_list = []
+
+         order_list = list(range(1, len(input_txt)+1))
+
+         if self.features is None:
```

Comment:
This check can move to `__init__`; that way it only runs once. (Currently it will run again each time `_extract` is called.)